x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview_header/engine_overview_header.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { EngineOverviewHeader } from '../engine_overview_header';
+import { KibanaContext } from '../../..';
+
+describe('EngineOverviewHeader', () => {
+  let enterpriseSearchUrl;
+
+  afterEach(() => {
+    enterpriseSearchUrl = undefined;
+  });
+
+  const render = () => {
+    return mount(
+      <KibanaContext.Provider
+        value={{
+          http: {},
+          enterpriseSearchUrl,
+          setBreadcrumbs: jest.fn(),
+        }}
+      >
+        <EngineOverviewHeader />
+      </KibanaContext.Provider>
+    );
+  };
+
+  describe('when enterpriseSearchUrl is set', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      enterpriseSearchUrl = 'http://localhost:3002';
+      wrapper = render();
+    });
+
+    describe('the Launch App Search button', () => {
+      const subject = () => wrapper.find('EuiButton[data-test-subj="launch_button"]');
```

Comment:
This is a very minor thing, but I would actually slightly prefer that we remain as library-agnostic as possible and instead just do `wrapper.find('[data-test-subj="launch_button"]')`. That way in the very unlikely scenario where we move away from EUI (or EUI changes their names?), we'll have less static strings to change, but primarily it's a best practice thing and one we also use in our E2E/Cypress tests.

As an alternative note - `snake_casing` is primarily for file/folder names in Kibana I believe, I believe we should be using `camelCase` for data attrs

Reply:
`wrapper.find('[data-test-subj="launch_button"]')` actually finds two elements for some reason, I haven't worked out exactly why, because there is only one rendered element in the `.html()` output. `'EuiButton[data-test-subj="launch_button"]'` was the only way I could properly select the element I was looking for. So this was basically a shortcut because I didn't feel like going any deeper into this problem right now. Happy to take a second look though.

Reply:
Interesting, I'm seeing the same thing. I think it's because EuiButton is passing `data-test-subj` down to as a prop to the raw `<button>` component, but the way Enzyme works, the prop is available on both the parent EuiButton component and the button child.

I'm fairly certain this wouldn't be an issue with shallow, so I'm good with leaving it for now and come back to it later when we've investigated using shallow more thoroughly.

If we could change to camelCasing though that would be dope 👍

Reply:
Dope. https://github.com/constancecchen/kibana/pull/4/commits/5f28efe392a5b59aa1fd64cd537183d5687f905c